test/fakeintake/server/body.go

```diff
@@ -15,15 +21,45 @@ type flareResponseBody struct {
 	Error  string `json:"error,omitempty"`
 }
 
-// getResponseBodyFromURLPath returns the appropriate response body to HTTP request sent to 'urlPath'
-func getResponseBodyFromURLPath(urlPath string) interface{} {
-	var body interface{}
+func getConnectionsResponse() []byte {
+	clStatus := &agentmodel.CollectorStatus{
+		ActiveClients: 1,
+		Interval:      30,
+	}
+	response := &agentmodel.ResCollector{
+		Message: "",
+		Status:  clStatus,
+	}
+	out, err := agentmodel.EncodeMessage(agentmodel.Message{
+		Header: agentmodel.MessageHeader{
+			Version:        agentmodel.MessageV3,
+			Encoding:       agentmodel.MessageEncodingProtobuf,
+			Type:           agentmodel.TypeResCollector,
+			OrgID:          503,
+			SubscriptionID: 2,
+		}, Body: response})
+	if err != nil {
+		panic(err) // will happen when new Message version exist and mark encodeV3 as retired
+	}
+	return out
+}
+
+// getResponseFromURLPath returns the appropriate response body to HTTP request sent to 'urlPath'
+func getResponseFromURLPath(urlPath string) httpResponse {
+	var response httpResponse
 
-	if urlPath == "/support/flare" {
-		body = flareResponseBody{CaseID: 0, Error: ""}
-	} else {
-		body = errorResponseBody{Errors: []string{}}
+	switch urlPath {
+	case "/support/flare":
+		response = buildSuccessResponse(flareResponseBody{CaseID: 0, Error: ""})
+	case "/api/v1/connections":
+		response = httpResponse{
+			statusCode:  http.StatusOK,
+			contentType: "application/x-protobuf",
+			body:        getConnectionsResponse(),
+		}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Wondering if we should rather have a per `urlPath` dictionary pointing to functions to build the complete response. I can see that the only part each response has in common is the status code, which is not a huge factorisation. 
Maybe having `getConnectionResponse` and `flareResponse` returning the full response and here looking in such a dictionary, and fall back to the get generic success response when not found

Reply:
I don't think this would worth the effort as you would still define different callback for each endpoint.
And probably the next (new) endpoint implementation will have a different response format

Reply:
en fait ce racourcci ne marche pas parceque dans `buildResponse(body, ...)`
`response.body = json.Marshal(body)`

Reply:
`et buildResponse() `change the  `content-type` in case of error

Reply:
This will mix both implmentation and a little bit cleaner

```
// getResponseFromURLPath returns the appropriate response body to HTTP request sent to 'urlPath'
func getResponseFromURLPath(urlPath string) httpResponse {
	var defaultResponse = httpResponse{
		statusCode:  http.StatusOK,
		contentType: "application/json",
		data:        errorResponseBody{Errors: []string{}},
	}

	responses := map[string]httpResponse{
		"/support/flare": {
			statusCode:  http.StatusOK,
			contentType: "application/json",
			data:        flareResponseBody{CaseID: 0, Error: ""},
		},
		"/api/v1/connections": {
			statusCode:  http.StatusOK,
			contentType: "application/x-protobuf",
			data:        getConnectionsResponse(),
		},
	}

	if _, found := responses[urlPath]; !found {
		return defaultResponse
	}
	return updateResponseFromData(responses[urlPath])
}

```